mcs/build/library.make

```diff
@@ -307,19 +307,25 @@ else
 GENSOURCES_RUNTIME=MONO_PATH="$(GENSOURCES_LIBDIR)$(PLATFORM_PATH_SEPARATOR)$$MONO_PATH" $(RUNTIME)
 endif
 
-sourcefile = $(depsdir)/$(PROFILE_PLATFORM)_$(PROFILE)_$(LIBRARY_SUBDIR)_$(LIBRARY).sources
+ifndef PROFILE_PLATFORM
```

Comment:
It's needed to be able to have host-specific (macos, win32, linux) basic/build profiles (when PROFILE_PLATFORM is not set yet)

Reply:
This seems ok but i'm not certain how it will interact with genproj. If it only applies to basic and build it should have no impact since we don't use those profiles in VS

Reply:
Though, shouldn't the =$(PROFILE_PLATFORM) be in ifdef, not ifndef?